scout/adapter/mongo/case.py

```diff
@@ -391,16 +368,89 @@ def cases(
 
         return self.case_collection.find(query).sort("updated_at", -1)
 
+    def last_modified_cases(
+        self, within_days, has_causatives, finished, reruns, research_requested, status
+    ):
+        """Retrieve cases that have been modified during the last n. days
+        Args:
+            within_days(int)
+            has_causatives(bool)
+            finished(bool)
+            reruns(bool)
+            research_requested(bool)
+            status(str)
+
+        Returns:
+            recent_cases(list): list of case _ids
+        """
+        verbs = set()
+        if has_causatives:
+            verbs.add("mark_causative")
+        if finished:
+            verbs.add("archive")
+            verbs.add("mark_causative")
+        if reruns:
+            verbs.add("rerun")
+        if research_requested:
+            verbs.add("open_research")
+        if status and status == "solved":
+            verbs.add("mark_causative")
+        verbs = list(verbs)
+
+        days_datetime = datetime.datetime.now() - datetime.timedelta(days=within_days)
+        # Look up 'mark_causative' events added since specified number days ago
+        event_query = {
+            "category": "case",
+            "verb": {"$in": verbs},
+            "created_at": {"$gte": days_datetime},
+        }
+        recent_events = self.event_collection.find(event_query)
+        recent_cases = set()
+        # Find what cases these events concern
+        for event in recent_events:
+            recent_cases.add(event["case"])
+        return list(recent_cases)
+
+    def sanger_ordered_cases(self, institute_id):
+        """Fetch all cases with at least a variant with validation ordered but not performed
+        Args:
+            institute_id(str): id of an institute
+
+        Returns:
+            sanger_missing(set): a set of case _ids with variants having Sanger validation missing
+        """
+        sanger_missing = set()
+        match_query = {
+            "$match": {
+                "institute": institute_id,
+                "sanger_ordered": True,
+                "$or": [
+                    {"validation": {"$nin": ["True positive", "False positive"]}},
+                    {"validation": {"$exists": False}},
+                ],
+            }
+        }
+        group = {
+            "$group": {
+                "_id": {
+                    "case": "$case_id",
+                },
+            }
+        }
+        pipeline = [match_query, group]
+        for res in self.variant_collection.aggregate(pipeline):
```

Comment:
Ho, didn't you also suggest removing the index for sanger_ordered on the variant collection? This is bound to get very slow if run without index. Wouldn't it be better to ask the the event collection first, and confirm the current status with the variants that have ever had Sanger order events?

Reply:
Compare the current adapter `VariantEventHandler sanger_ordered()` which does basically this?

Reply:
I agree with you that having also VariantEventHandler sanger_ordered() is confusing. But they are different in this:

- this function collects the sanger ordered but not validated - the other one is collecting the sanger ordered only (could be validated or not)
- this function is collecting variant objects (from variant collection - I've reintroduced the index in this PR, btw :) ) and not variant IDs
- this function is not selecting the sanger ordered vars for a user - the other does it.

I'll try to modify the code to make it more clear (and eventually remove redundant code!)

Reply:
Hehe, I didn't see that, sorry! I may not have reloaded this morning. Still, it is a fairly big index if it is only for this?

Right, there are certainly things that are specific here; just saying going via the events (small, indexed) would be faster than unindexed variants. I guesstimate it would be faster doing that plus checking status than searching indexed variants. But, the index does exist so if you must.. 🤷

Reply:
Fixed by re-using the existing function!